src/pdb/mod.rs

```diff
@@ -381,8 +450,8 @@ pub struct RowGroup {
     /// An offset which points to a row in the table, whose actual presence is controlled by one of the
     /// bits in `row_present_flags`. This instance allows the row itself to be lazily loaded, unless it
     /// is not present, in which case there is no content to be loaded.
-    #[br(offset = page_heap_offset, args { count: num_rows.into(), inner: (page_type,) })]
-    rows: Vec<FilePtr16<Row>>,
+    #[br(offset = page_heap_offset, parse_with = FilePtr16::parse, args { count: num_rows.into(), inner: (page_type,) })]
+    rows: Vec<Row>,
```

Comment:
This is not the same and thus this branch also broke plain reading unfortunately. The proper implementation is a vector of pointers while the new one is a pointer of a vector. 

Also, since the manual write implementation `RowGroup::write_options_and_get_row_offset` is already quite complex I'm questioning it makes sense to continue down this path. In the end it comes down to requiring some sort of `FilePtr` serialization IMO. Also, some sort of abstract heap would be nice too (since Pages are also essentially just heap allocated too).

Reply:
Well,  that would require computing the offsets beforehand (to instantiate `FilePtr32`), which would make it hard to create a PDB file from scratch. The user likely does not care about the offsets. This is why I chose this approach, but I'm open to ideas.